### ISSUE TITLE
problem with go mod

### ISSUE THREAD

**Issue by vtolstov** [nats-io/nats-streaming-server#737]
can you add go.mod to repo?
```
github.com/nats-io/nats-streaming-server/server
# github.com/nats-io/nats-streaming-server/server
vendor/github.com/nats-io/nats-streaming-server/server/clustering.go:276:28: cannot use s.log (type *"github.com/nats-io/nats-streaming-server/logger".StanLogger) as type "github.com/nats-io/nats-streaming-server/logger".Logger in argument to newRaftLog:
        *"github.com/nats-io/nats-streaming-server/logger".StanLogger does not implement "github.com/nats-io/nats-streaming-server/logger".Logger (missing Warnf method)
vendor/github.com/nats-io/nats-streaming-server/server/server.go:872:7: cannot use ss.stan.log (type *"github.com/nats-io/nats-streaming-server/logger".StanLogger) as type "github.com/nats-io/nats-streaming-server/logger".Logger in assignment:
        *"github.com/nats-io/nats-streaming-server/logger".StanLogger does not implement "github.com/nats-io/nats-streaming-server/logger".Logger (missing Warnf method)
vendor/github.com/nats-io/nats-streaming-server/server/server.go:1451:37: cannot use s.log (type *"github.com/nats-io/nats-streaming-server/logger".StanLogger) as type "github.com/nats-io/nats-streaming-server/logger".Logger in argument to stores.NewFileStore:
        *"github.com/nats-io/nats-streaming-server/logger".StanLogger does not implement "github.com/nats-io/nats-streaming-server/logger".Logger (missing Warnf method)
vendor/github.com/nats-io/nats-streaming-server/server/server.go:1454:36: cannot use s.log (type *"github.com/nats-io/nats-streaming-server/logger".StanLogger) as type "github.com/nats-io/nats-streaming-server/logger".Logger in argument to stores.NewSQLStore:
        *"github.com/nats-io/nats-streaming-server/logger".StanLogger does not implement "github.com/nats-io/nats-streaming-server/logger".Logger (missing Warnf method)
vendor/github.com/nats-io/nats-streaming-server/server/server.go:1457:39: cannot use s.log (type *"github.com/nats-io/nats-streaming-server/logger".StanLogger) as type "github.com/nats-io/nats-streaming-server/logger".Logger in argument to stores.NewMemoryStore:
        *"github.com/nats-io/nats-streaming-server/logger".StanLogger does not implement "github.com/nats-io/nats-streaming-server/logger".Logger (missing Warnf method)
vendor/github.com/nats-io/nats-streaming-server/server/server.go:4421:18: cannot use s.log (type *"github.com/nats-io/nats-streaming-server/logger".StanLogger) as type "github.com/nats-io/nats-streaming-server/logger".Logger in argument to traceSubState:
        *"github.com/nats-io/nats-streaming-server/logger".StanLogger does not implement "github.com/nats-io/nats-streaming-server/logger".Logger (missing Warnf method)
```

after fix go mod to gnats v1.3.0 i have error
```
vendor/github.com/nats-io/nats-streaming-server/server/monitor.go:207:10: cannot use limits.OpenFiles (type int64) as type int in assignment
```

**Comment by vtolstov** [comment]
@kozlovic can you fix the vendor/github.com/nats-io/nats-streaming-server/server/monitor.go:207:10: cannot use limits.OpenFiles (type int64) as type int in assignment ?

**Comment by kozlovic** [comment]
In my vendor dir, limits.OpeFiles is `int`, not `int64`.

**Comment by vtolstov** [comment]
thanks, i'm try to update procfs, but if you add go.mod this is not needed...

**Comment by vtolstov** [comment]
https://godoc.org/github.com/prometheus/procfs#ProcLimits in master this int64

**Comment by vtolstov** [comment]
as i see in procfs this change realy old
```
commit a6e9df898b1336106c743392c48ee0b71f5c4efa
Merge: e645f4e5aaa8 9f8f30874832
Author: Tobias Schmidt <tobidt@gmail.com>
Date:   Tue Oct 17 23:40:25 2017 +0200

    Merge pull request #64 from ncabatoff/support-64bit-limits

    Fix #62 by switching all limits to int64
```
